drivers/storage/portworx/util/util.go

```diff
@@ -1902,3 +1905,97 @@ func isVersionSupported(current, target string) bool {
 
 	return currentVersion.Core().GreaterThanOrEqual(targetVersion)
 }
+
+// Get list of storagenodes that are a part of the current cluster that need a node PDB
+func NodesNeedingPDB(cluster *corev1.StorageCluster, sdkConn *grpc.ClientConn, k8sClient client.Client) ([]string, error) {
+	// Get the list of storage nodes
+	nodeClient := api.NewOpenStorageNodeClient(sdkConn)
+	ctx, err := SetupContextWithToken(context.Background(), cluster, k8sClient)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeEnumerateResponse, err := nodeClient.EnumerateWithFilters(
+		ctx,
+		&api.SdkNodeEnumerateWithFiltersRequest{},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to enumerate nodes: %v", err)
+	}
+
+	// Get the list of k8s nodes that are part of the current cluster
+	k8sNodesStoragePodCouldRun := make(map[string]bool)
+	k8sNodeList := &v1.NodeList{}
+	err = k8sClient.List(context.TODO(), k8sNodeList)
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range k8sNodeList.Items {
+		shouldRun, shouldContinueRunning, err := k8sutil.CheckPredicatesForStoragePod(&node, cluster, nil)
+		if err != nil {
+			return nil, err
+		}
+		if shouldRun || shouldContinueRunning {
```

Comment:
Why do we need to check if that k8s node can run a portworx pod or not? Don't we need to only check if the portworx node is part of the current cluster or not? Basically this can be a simple map - 
```
for _, node := range k8sNodeList.Items {
  k8sNodes[node.Name] = true
}
```

Reply:
It was mentioned that we will create PDB only for k8s nodes that can run px right? So during surge upgrade when a node is being removed from the cluster, it shouldn't run px and we wont create a node PDB for it .

Reply:
For now I will make the change suggested, but lets discuss why we need to create node PDB for k8s nodes that should not run px pod

Reply:
If a portworx node has scheduler name which is part of the cluster, then we should protect that k8s node from upgrading irrespective of whether currently the k8s node should run portworx or not, right?

Let's discuss more on this offline to make sure we are covering all cases correctly.

Reply:
I assumed a kubernetes node when it goes to the state of should not run portworx, then px will be deleted from that node soon anyways, so why should we create pdb for it only to delete it in sometime?